lib/modules/model/types/ModelContent.ts

```diff
@@ -33,16 +33,27 @@ export interface LocaleDetails {
   description: string;
 }
 
-interface DefinitionMetadataDetails {
+interface BaseDefinition {
   readOnly?: boolean;
-  type?: string;
-  values?: string[] | number[] | boolean[];
+  type: string;
+}
+
+export enum MetadataDetailsEnum {
+  OPTION_SELECTOR = "optionSelector",
+  DATETIME = "datetime",
+}
+
+export interface OptionsSelectorDefinition extends BaseDefinition {
+  type: typeof MetadataDetailsEnum.OPTION_SELECTOR;
```

Comment:
`typeof` is not necessary here, `typeof MetadataDetailsEnum.OPTION_SELECTOR === MetadataDetailsEnum.OPTION_SELECTOR`